collector/internal/telemetryapi/listener.go

```diff
@@ -132,13 +156,8 @@ func (s *Listener) Wait(ctx context.Context, reqID string) error {
 					s.logger.Warn("non-Event found in queue", zap.Any("item", item))
 					continue
 				}
-				s.logger.Debug("Event processed", zap.Any("event", i))
-				if i.Type != "platform.runtimeDone" {
-					continue
-				}
-
-				if i.Record["requestId"] == reqID {
-					return nil
+				if err := s.handleEvent(ctx, i); err != nil {
+					return err
```

Comment:
Should this return on error?  I think this function will only get called once and thus returning on error here will prevent any further events from processing.  Does this eventually result in the collector extension being terminated?